### PR TITLE
Translate to Spanish

### PR DESCRIPTION
**Pull request by DavidMedinaO**
Translate to Spanish:
template method pattern
throttling pattern
trampoline pattern
transaction-script pattern
type object pattern
Update method pattern
visitor pattern

### CODE REVIEW

File: localization/es/throttling/README.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+  public void reset() {
+    tenantCallsCount.replaceAll((k, v) -> new AtomicLong(0));
+    LOGGER.info("reset counters");
+  }
+}
+```
+
+A continuación, se introduce el servicio al que llaman los inquilinos. Para realizar un seguimiento del número de llamadas, se utiliza un temporizador de estrangulamiento.
+temporizador.
```
Comment (luismateoh) [inline localization/es/throttling/README.md:80]: 'temporizador.' is left over

File: localization/es/trampoline/README.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+El patrón Trampoline se utiliza para implementar algoritmos recursivamente en Java sin volar la pila
+y para intercalar la ejecución de funciones sin codificarlas juntas.
+
+## Explicación
+
+La recursión es una técnica frecuentemente adoptada para resolver problemas algorítmicos en un estilo de divide y vencerás.
+estilo divide y vencerás. Por ejemplo, el cálculo de la suma acumulativa de Fibonacci y los factoriales. En este tipo de
+problemas, la recursividad es más directa que su homóloga de bucle. Además, la recursividad puede
+necesitar menos código y parecer más concisa. Hay un dicho que dice que todo problema de recursión puede resolverse
+utilizando un bucle a costa de escribir código más difícil de entender.
```
Comment (luismateoh) [inline localization/es/trampoline/README.md:20]: review duplicated sentences

File: localization/es/trampoline/README.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+En palabras sencillas
+
+> El patrón Trampoline permite la recursión sin agotar la memoria de la pila.
+
+Wikipedia dice
+
+> En Java, trampoline se refiere al uso de reflection para evitar el uso de clases internas, por ejemplo en event
+> eventos. La sobrecarga de tiempo de una llamada a reflection se intercambia por la sobrecarga de espacio de una clase interna.
+> clase interna. Trampolines en Java generalmente implican la creación de un GenericListener para pasar eventos a una clase externa.
+> una clase externa.
```
Comment (luismateoh) [inline localization/es/trampoline/README.md:42]: Duplicated sentences

File: localization/es/throttling/README.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+
+En palabras sencillas
+
+> El patrón de Throttling se utiliza para limitar la velocidad de acceso a un recurso.
+
+[Microsoft documentation](https://docs.microsoft.com/en-us/azure/architecture/patterns/throttling) dice
+
+> Controlar el consumo de recursos utilizados por una instancia de una aplicación, un tenant individual,
+> o un servicio completo. Esto puede permitir que el sistema continúe funcionando y cumpla con los acuerdos de nivel de servicio, incluso cuando un aumento de la demanda impone una carga extrema sobre los recursos.
+> acuerdos de nivel de servicio, incluso cuando un aumento de la demanda impone una carga extrema sobre los recursos.
```
[the remark this answers is not in the document]
Comment (DavidMedinaO) [reply localization/es/throttling/README.md]: Done

File: localization/es/throttling/README.md (hunk 1 above)
Comment (DavidMedinaO) [reply localization/es/throttling/README.md]: Done

File: localization/es/template-method/README.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+Ejemplo del mundo real
+
+> Los pasos generales para robar un objeto son los mismos. Primero, eliges al objetivo, luego lo confundes...
+> y finalmente, robas el objeto. Sin embargo, hay muchas maneras de implementar estos
+> pasos.
+
+En palabras sencillas
+
+> El patrón Template Method esboza los pasos generales en la clase padre y deja que las implementaciones hijas concretas definan los detalles.
+> definan los detalles.
```
[the remark this answers is not in the document]
Comment (DavidMedinaO) [reply localization/es/template-method/README.md]: Done

File: localization/es/trampoline/README.md (hunk 2 above)
Comment (DavidMedinaO) [reply localization/es/trampoline/README.md]: Done

File: localization/es/trampoline/README.md (hunk 3 above)
Comment (DavidMedinaO) [reply localization/es/trampoline/README.md]: Done

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Looks good! Thank you for the contribution 🎉

@all-contributors please add @DavidMedinaO for translation

**Comment by iluwatar** [comment]
@all-contributors please add @luismateoh for review

**Pull request merged**